src/Compute/Compute/Strategies/ComputeRp/VirtualMachineStrategy.cs

```diff
@@ -116,16 +116,23 @@ static class VirtualMachineStrategy
                         },
                         StorageProfile = new StorageProfile
                         {
-                            //ImageReference = (imageReferenceId.Contains("CommunityGalleries")) ? new ImageReference { CommunityGalleryImageId = imageReferenceId}
-                            ImageReference = (imageReferenceId == null) ? imageAndOsType?.Image : (imageReferenceId.ToLower().StartsWith("/communitygalleries/") ? new ImageReference
-                            {
-                                CommunityGalleryImageId = imageReferenceId,
-                                SharedGalleryImageId = sharedGalleryImageId
-                            }: new ImageReference
-                            {
-                                Id = imageReferenceId,
-                                SharedGalleryImageId = sharedGalleryImageId
-                            }),
+                            ImageReference = (imageReferenceId == null && sharedGalleryImageId == null) ? imageAndOsType?.Image
+                                : (sharedGalleryImageId != null ? new ImageReference
+                                {
+                                    SharedGalleryImageId = sharedGalleryImageId
+                                }
+                                : (imageReferenceId.ToLower().StartsWith("/communitygalleries/") ? new ImageReference
```

Comment:
same null concern in this block of code and multiple if checks.

Reply:
First condition checks if {{both imagereferenceId and sharedgalleryimageid} are null
{ ELSE (implies one of them is not null)
{ Check2 : sharedgalleryimage is not null 
{ ELSE (shared image gallery is null hence imagereference is not null)
}}}

Also tested this